src/lib/UXDController.ts

```diff
@@ -142,44 +156,50 @@ export class UXDController {
   }: {
     contractAddress: string;
     spender: string;
-    amount: BigNumber;
+    amount: number;
     signer: Signer;
   }): Promise<ContractTransaction> {
-    return ERC20__factory.connect(contractAddress, signer).connect(signer).approve(spender, amount);
+    const ethAmount = ethers.utils.parseEther(amount.toString());
+    return ERC20__factory.connect(contractAddress, signer).connect(signer).approve(spender, ethAmount);
   }
 
-  public allowance({
+  public async allowance({
     contractAddress,
     account,
     spender,
   }: {
     contractAddress: string;
     account: string;
     spender: string;
-  }): Promise<BigNumber> {
-    return ERC20__factory.connect(contractAddress, this.provider).allowance(account, spender);
+  }): Promise<number> {
+    const allowance = await ERC20__factory.connect(contractAddress, this.provider).allowance(account, spender);
+    return Number(ethers.utils.formatEther(allowance));
   }
 
-  public tokenBalance({
+  public async tokenBalance({
     contractAddress,
     account,
   }: {
     contractAddress: string;
     account: string;
-  }): Promise<BigNumber> {
-    return ERC20__factory.connect(contractAddress, this.provider).balanceOf(account);
+  }): Promise<number> {
+    const balance = await ERC20__factory.connect(contractAddress, this.provider).balanceOf(account);
+    return Number(ethers.utils.formatEther(balance));
   }
 
-  public uxdTotalSupply(): Promise<BigNumber> {
-    return this.uxdContract.totalSupply();
+  public async uxdTotalSupply(): Promise<number> {
+    const totalSupply = await this.uxdContract.totalSupply();
+    return Number(ethers.utils.formatEther(totalSupply));
   }
 
-  public mintedPerCollateral(token: string): Promise<BigNumber> {
-    return this.controllerContract.mintedPerCollateral(token);
+  public async mintedPerCollateral(token: string): Promise<number> {
+    const minted = await this.controllerContract.mintedPerCollateral(token);
+    return Number(ethers.utils.formatEther(minted));;
   }
 
-  public redeemable(token: string): Promise<BigNumber> {
-    return this.controllerContract.redeemable(token);
+  public async getRedeemableCollateral(token: string): Promise<number> {
```

Comment:
`getRedeemableAmountForCollateral` ?

Reply:
This is the amount of collateral you can redeem.
It is much collateral has been deposited? For now it'll just be the amount of WETH (ETH + WETH) but we're tracking both asWETH.

Reply:
Is it the amount of collateral backing the delta-neutral position, or the "depository" balance for the collateral?

Reply:
Change to `getDepositedCollateralAmount(token)`